version/version.go

```diff
@@ -17,6 +17,13 @@ var MICVersion string
 // NMIVersion is the version of the NMI component
 var NMIVersion string
 
+// GetUserAgent is used to get the user agent string which is then provided to adal
+// to use as the extended user agent header.
+// The format is: aad-pod-identity/<component, either NMI or MIC>/<Version of MIC>/<Version of NMI>/<Git commit>/<Build date>
+func GetUserAgent(component string) string {
+	return fmt.Sprintf("aad-pod-identity/%s/%s/%s/%s/%s", component, MICVersion, NMIVersion, GitCommit, BuildDate)
```

Comment:
This ua string seems strange.
Why not pass the version to use as a function argument, or store it in a map as a package var?